### DISCUSSION TITLE
Cursor not working with touch-screen

### DISCUSSION THREAD

**Discussion by OrangeChef** [ppy/osu#16840] [Q&A] [answered]
I just tried out Osu! lazer earlier today on my laptop. I found out that when using the touch-screen, the cursor jumps to somewhere in the top-left corner and moves very slowly. It seems to be impossible to make it move to the center or anywhere else. Buttons and UI interactions also do not work because of this. In the video you can see that every time I tap the screen the cursor moves slightly, but remains in the corner. Is this a known-issue?

https://user-images.githubusercontent.com/75186386/153349593-f1b4aa77-52d8-4d0b-8274-a7cb5bdb711f.mp4

**Comment by frenzibyte** [comment, marked as answer]
Out of curiosity, does disabling "High precision mouse" resolve your issue? osu!lazer does not have support for sending events from touchscreen input natively yet.

**Reply by OrangeChef** [reply]
Wow, that actually worked... Pretty ironic that disabling high-precision made it more precise. Thanks!

**Reply by peppy** [reply]
Could you please give more details on the device/OS you are using? This looks like it will probably be a bug in our local implementation of raw input in o!f, which means we should look at a fix for it.

**Reply by OrangeChef** [reply]
I am running on Windows 11 22494.1000. Touch-screen works perfectly on the stable client, and works fine on lazer with high-precision disabled.

**Reply by peppy** [reply]
I'm more interested in which touch screen are you using, ie. is it a surface or something else?

**Reply by OrangeChef** [reply]
Oh, the only info I have on the screen itself is from a sticker that says, "15.6" HD True Life Touch Display."
If you also want the laptop model, it is a DELL Inspiron 15 3000.

**Reply by peppy** [reply]
Thanks. I'll test with the hardware I have available here (a surface gen 4) and see if I can reproduce the same issue. It's either going to be an issue with raw input on all desktop touch screens, or an input specific to your hardware/driver combo. If it's the former it's definitely something we want to fix.

**Reply by peppy** [reply]
Can reproduce the failure on my surface, so will investigate this further.

**Reply by peppy** [reply]
Tracking at https://github.com/ppy/osu-framework/issues/5033.
